### PR TITLE
ENH: adding safety try-catch blocks to prevent whole test suite from falling over

### PR DESCRIPTION
**Pull request by IgorNadj**
Every once in a while something falls over due to the inherent unreliability of webdriver and the browser, and this catches it and lets the test suite continue to the next scenario rather than just falling over completely. 

-- P.S. just found this handy tip too, you can do a compare on github ignoring whitespace by adding ?w=1, e.g. https://github.com/silverstripe-labs/silverstripe-behat-extension/compare/master...IgorNadj:patch-safety?w=1

### CODE REVIEW

File: src/SilverStripe/BehatExtension/Context/BasicContext.php [hunk 1]
```diff
@@ -865,4 +899,14 @@ public function iScrollToElement($locator) {
...
 		$this->getSession()->executeScript($js);
 	}
+	
+	
+	
+	/**
+	 * We have to catch exceptions and log somehow else otherwise behat falls over
+	 */
+	protected function logException($e){
+		file_put_contents('php://stderr', 'Exception caught: '.$e->getMessage());
```
Comment (chillu) [inline src/SilverStripe/BehatExtension/Context/BasicContext.php]: We need the stacktrace for efficient debugging here as well, right?

File: src/SilverStripe/BehatExtension/Context/BasicContext.php [hunk 2]
```diff
@@ -387,16 +411,26 @@ public function iTypeIntoTheDialog($data) {
...
      */
 	public function iConfirmTheDialog() {
-        $this->getSession()->getDriver()->getWebDriverSession()->accept_alert();
-        $this->handleAjaxTimeout();
+		try{
+			$this->getSession()->getDriver()->getWebDriverSession()->accept_alert();
+			$this->handleAjaxTimeout();
+		}catch(\WebDriver\Exception $e){
+			// have to catch and log somehow else otherwise behat result parsing fails
+			assertTrue(false, 'Dialog bug: '.$e->getMessage());
```
Comment (chillu) [inline src/SilverStripe/BehatExtension/Context/BasicContext.php]: Huh? Can you show how it fails? I'd rather avoid this and call `logException()` as in other places. Plus, removing the tack trace here (and having them same message in both `iConfirmTheDialog()` and `iDismissTheDialog()` will make debugging harder.

**Comment by chillu** [comment]
Looks like a good improvement in general, thanks Igor! Can you please rebase?

File: src/SilverStripe/BehatExtension/Context/BasicContext.php (hunk 2 above)
Comment (IgorNadj) [inline src/SilverStripe/BehatExtension/Context/BasicContext.php]: Whoops, this should not be here. Unrelated issue.

File: src/SilverStripe/BehatExtension/Context/BasicContext.php (hunk 1 above)
Comment (IgorNadj) [inline src/SilverStripe/BehatExtension/Context/BasicContext.php]: Yep, makes sense.

**Comment by IgorNadj** [comment]
Cool, have cleaned things up and rebased

File: src/SilverStripe/BehatExtension/Context/BasicContext.php [hunk 3]
```diff
@@ -942,4 +966,14 @@ public function spin($lambda, $wait = 60) {
...
         ));
     }
+	
+	
+	
+	/**
+	 * We have to catch exceptions and log somehow else otherwise behat falls over
+	 */
+	protected function logException($e){
+		file_put_contents('php://stderr', 'Exception caught: '.$e);
```
Comment (chillu) [inline src/SilverStripe/BehatExtension/Context/BasicContext.php]: Right, confirmed that toString() does something useful: http://php.net/manual/en/exception.tostring.php - good stuff.

**Comment by chillu** [comment]
Thanks Igor!

**Pull request merged**